playbooks/nvidia-software/nvidia-mig.yml

```diff
@@ -1,120 +1,52 @@
 ---
-# Enable/disable MIG mode
-# run with tags --enable or --disable
-- hosts: all
-  become: yes
+# Install the NVIDIA MIG Manager tooling on all MIG-capable nodes
+# Copy over the custom MIG config to all nodes and apply them
+# Cluster-wide config is set in group_vars/config.yml by mig_manager_profile
+# Per-node config can be configured in the inventory file
 
+# Check if MIG capabilities and software on nodes
+- hosts: all
   vars:
-    deepops_mig_devices: "all"
-    nv_services:
-      - nvsm
-      - nvidia-persistenced
-      - nvidia-fabricmanager
-      - nv_peer_mem
-      - dcgm
-      - docker
-    nv_modules:
-      - nv_peer_mem
-      - nvidia_uvm
-      - nvidia_drm
-      - nvidia_modeset
-      - nvidia
-
+    mig_manager_reboot_timeout: 900
   tasks:
-    # Check for MIG-capable devices
     - name: check for MIG capable devices
-      command: nvidia-smi --query-gpu=pci.bus_id,mig.mode.current --format=csv,noheader
+      shell: nvidia-smi --query-gpu=mig.mode.current --format=csv,noheader | grep -v 'N/A'
       register: has_mig
+      failed_when: false
 
-    # Pre-tasks
-    - name: stop system services
-      systemd:
-        state: stopped
-        enabled: no
-        name: "{{ item }}"
-      with_items: "{{ nv_services }}"
-      tags: enable, disable, never
-    - name: unload drivers
-      modprobe:
-        state: absent
-        name: "{{ item }}"
-      with_items: "{{ nv_modules }}"
-      tags: enable, disable, never
-
-    # Manage MIG
-    - name: enable MIG mode (all devices)
-      command: nvidia-smi -mig 1
-      tags: enable, never
-      when: deepops_mig_devices | default("") == "all"
-    - name: enable MIG mode (per device)
-      command: nvidia-smi -mig 1 -i "{{ deepops_mig_devices }}"
-      tags: enable, never
-      when: deepops_mig_devices | default("") != "" and  deepops_mig_devices | default("") != "all"
+    - name: Install MIG Manager
+      include_role:
+        name: nvidia-mig-manager
+      when:
+        - has_mig.rc == 0
+    
+    - name: copy cluster-wide mig config file
+      copy:
+        src: "../../config/nvidia-mig-config.yml"
+        dest: "{{ mig_manager_config }}"
+      when: has_mig.rc == 0
 
-    - name: disable MIG mode (all devices)
-      command: nvidia-smi -mig 0
-      tags: disable, never
-      when: deepops_mig_devices | default("") == "all"
-    - name: disable MIG mode (per device)
-      command: nvidia-smi -mig 0 -i "{{ deepops_mig_devices }}"
-      tags: disable, never
-      when: deepops_mig_devices | default("") != "" and  deepops_mig_devices | default("") != "all"
+    - name: Apply MIG configuration
+      command: nvidia-mig-parted apply -f {{ mig_manager_config }} -c {{ mig_manager_profile }}
```

Comment:
After discussion in an internal bug, this line should be:

```
command: nvidia-mig-parted apply -f {{ mig_manager_config }} -c {{ mig_manager_profile }} -k /etc/nvidia-mig-manager/hooks.yaml
```

Where you might choose to put the hooks file in a variable as well. :)